lib/public-stats.js

```diff
@@ -227,3 +228,44 @@ const updateDailyDealsStats = async (pgClient, committees, findDealClients) => {
     flatStats.map(stat => stat.retrievable)
   ])
 }
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ */
+const updateRetrievalTimings = async (pgClient, committees) => {
+  /** @type {Map<string, number[]>} */
+  const retrievalTimings = new Map()
+  for (const c of committees) {
+    if (!c.evaluation || !c.evaluation.hasRetrievalMajority || c.evaluation.retrievalResult !== 'OK') continue
+    const { minerId } = c.retrievalTask
+    const ttfbMeasurments = []
+    for (const m of c.measurements) {
+      if (m.fraudAssessment !== 'OK' || !m.first_byte_at || !m.start_at) continue
+      ttfbMeasurments.push(m.first_byte_at - m.start_at)
+    }
+
+    if (!retrievalTimings.has(minerId)) {
+      retrievalTimings.set(minerId, [])
+    }
+
+    const ttfb = Math.ceil(getValueAtPercentile(ttfbMeasurments, 0.5))
+    retrievalTimings.get(minerId).push(ttfb)
+  }
+
+  // eslint-disable-next-line camelcase
+  const rows = Array.from(retrievalTimings.entries()).flatMap(([miner_id, ttfb_p50]) => ({ miner_id, ttfb_p50 }))
+
+  await pgClient.query(`
+    INSERT INTO retrieval_timings (day, miner_id, ttfb_p50) 
+    SELECT now(), miner_id, ttfb_p50 FROM jsonb_to_recordset($1::jsonb) AS t (miner_id text, ttfb_p50 int[])
+    ON CONFLICT (day, miner_id)
+    DO UPDATE SET
+      ttfb_p50 = array_cat(
+        retrieval_timings.ttfb_p50,
+        EXCLUDED.ttfb_p50
+      )
+  `, [
+    JSON.stringify(rows)
```

Comment:
I have done this as I am unable to `unnest` nested array (`unnest` would convert nested array to int values). If you have some suggestions on how to do this in a cleaner way I'd appreciate that 🙏🏻

Reply:
I've tried removing `JSON.stringify` but it throws following error `error: invalid input syntax for type json`. I think node-postgres handles serialization but only to postgres binary format, so we need to serialize it to json first before it get's converted to binary.